### DISCUSSION TITLE
How to combine relm4 with sourceview5?

### DISCUSSION THREAD

**Discussion by dlight** [Relm4/Relm4#494] [Q&A] [answered]
In the [examples](https://github.com/Relm4/Relm4/tree/main/examples) I seen so far, only widgets that come builtin in the gtk crate are used.

But can relm4 work with the [sourceview5](https://gitlab.gnome.org/World/Rust/sourceview5-rs/) crate? And more generally, can it work with any Gtk widget?

(Maybe a sourceview5 example would be useful)

**Comment by AaronErhardt** [comment, marked as answer]
Yes, it should just work with Relm4, like any other GTK-based library (e. g. libadwaita or libpanel also work with Relm4).

I don't have time to provide an example, but I'd be happy to hear about your experience with Relm4 + sourceview5.
